src/sparc/client/services/metadata.py

```diff
@@ -0,0 +1,278 @@
+        except requests.exceptions.Timeout as errt:
+            logging.error("Retrieving URL - Timeout Error:", errt)
+            success = 0
+        except requests.exceptions.RequestException as err:
+            logging.error("Retrieving URL - Something Else", err)
+            success = 0
+
+        url_session.close()
+
+        if success == 1:
+            result = url_result
+        else:
+            result = {}
+
+        return result.json()
+
+    #####################################################################
+    # Function to retrieve content via POST from URL with retries
+    def postURL(self, url, body, headers="NONE"):
+        result = "[ERROR]"
+        url_session = requests.Session()
+
+        retries = Retry(
+            total=6, backoff_factor=1, status_forcelist=[403, 404, 413, 429, 500, 502, 503, 504]
+        )
+
+        url_session.mount("https://", HTTPAdapter(max_retries=retries))
+
+        try:
+            if type(body) is dict:
+                body_json = body
+            else:
+                body_json = json.loads(body)
+        except:
+            logging.error("Elasticsearch query body can not be read")
+
+        success = 1
+
+        try:
+            if headers == "NONE":
+                url_result = url_session.post(url, json=body_json)
+            else:
+                url_result = url_session.post(url, json=body_json, headers=headers)
+
+            if url_result.status_code == 410:
+                logging.warning("Retrieval Status 410 - URL Unpublished:" + url)
+            else:
+                url_result.raise_for_status()
+
+        except requests.exceptions.HTTPError as errh:
+            logging.error("Retrieving URL - HTTP Error:", errh)
+            success = 0
+        except requests.exceptions.ConnectionError as errc:
+            logging.error("Retrieving URL - Error Connecting:", errc)
+            success = 0
+        except requests.exceptions.Timeout as errt:
+            logging.error("Retrieving URL - Timeout Error:", errt)
+            success = 0
+        except requests.exceptions.RequestException as err:
+            logging.error("Retrieving URL - Something Else", err)
+            success = 0
+
+        url_session.close()
+
+        if success == 1:
+            result = url_result
+        else:
+            result = {}
+
+        return result.json()
+
+    #####################################################################
+    # Metadata Search Functions
+
+    def list_datasets(self, limit: int = 10, offset: int = 0) -> list:
+        """Lists datasets and associated metadata.
+
+        Parameters:
+        -----------
+        limit : int
+            Max number of datasets returned.
+        offset : int
+            Offset used for pagination of results.
+
+        Returns:
+        --------
+        A json with the results.
+
+        """
+        self.host_api = "https://scicrunch.org/api/1/elastic/SPARC_Algolia_pr/_search"
```

Comment:
A bit confused with the previous definition (line 55)
        self.host_api = "https://scicrunch.org/api/1/elastic"
Is this ok? Maybe we should reuse `self.host_api` here?

Reply:
This is the full formed URL that is needed.  The one above is the stub that still needs the appropriate index (SPARC_Algolia_pr) and function (_search) added. This can change between methods.  So info is correct - that is the service we are connecting to.  And then with each method need to make sure we are usinig proper Index and method

Reply:
This needs a generator function you shouldn't be setting the host api all over the place like this.